### ISSUE TITLE
[Bug]: ValidationError: 1 validation error for DataSource component_type subclass of BaseComponent expected (type=type_error.subclass; expected_class=BaseComponent)

### ISSUE THREAD

**Issue by andysingal** [run-llama/llama_index#11534]
### Bug Description

```
ValidationError: 1 validation error for DataSource component_type subclass of BaseComponent expected (type=type_error.subclass; expected_class=BaseComponent)

Traceback:
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 535, in _run_script
    exec(code, module.__dict__)
File "/Users/ankushsingal/Desktop/llamaindex/app.py", line 6, in <module>
    from llama_index.core import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/__init__.py", line 13, in <module>
    from llama_index.callbacks.global_handlers import set_global_handler
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/callbacks/__init__.py", line 7, in <module>
    from .token_counting import TokenCountingHandler
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/callbacks/token_counting.py", line 6, in <module>
    from llama_index.utilities.token_counting import TokenCounter
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/utilities/token_counting.py", line 6, in <module>
    from llama_index.llms import ChatMessage, MessageRole
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/llms/__init__.py", line 3, in <module>
    from llama_index.llms.anyscale import Anyscale
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/llms/anyscale.py", line 9, in <module>
    from llama_index.llms.openai import OpenAI
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/llms/openai/__init__.py", line 1, in <module>
    from llama_index.llms.openai.base import AsyncOpenAI, OpenAI, SyncOpenAI, Tokenizer
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/llms/openai/base.py", line 16, in <module>
    from llama_index.core.base.llms.types import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/__init__.py", line 19, in <module>
    from llama_index.core.indices import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/indices/__init__.py", line 4, in <module>
    from llama_index.core.indices.composability.graph import ComposableGraph
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/indices/composability/__init__.py", line 4, in <module>
    from llama_index.core.indices.composability.graph import ComposableGraph
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/indices/composability/graph.py", line 7, in <module>
    from llama_index.core.indices.base import BaseIndex
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/indices/base.py", line 12, in <module>
    from llama_index.core.ingestion import run_transformations
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/__init__.py", line 2, in <module>
    from llama_index.core.ingestion.pipeline import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/pipeline.py", line 31, in <module>
    from llama_index.core.ingestion.api_utils import get_client
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/api_utils.py", line 20, in <module>
    from llama_index.core.ingestion.data_sources import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/data_sources.py", line 340, in <module>
    ConfigurableDataSources = build_conifurable_data_source_enum()
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/data_sources.py", line 117, in build_conifurable_data_source_enum
    DataSource(
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/pydantic/v1/main.py", line 341, in __init__
    raise validation_error
```

### Version

0.10.14

### Steps to Reproduce

```
import logging
import sys

import streamlit as st
import tiktoken
from llama_index.core import (
    PromptTemplate,
    ServiceContext,
    SimpleDirectoryReader,
    VectorStoreIndex,
)
from llama_index.core.indices.prompt_helper import PromptHelper
from llama_index.core.node_parser import SimpleNodeParser
from llama_index.embeddings.openai import OpenAIEmbedding
from llama_index.llms.openai import OpenAI

# load_dotenv()

# load_dotenv()
OPENAI_API_KEY = ""
# os.environ["OPENAI_API_KEY"]=os.getenv('OPENAI_API_KEY')

llm = OpenAI(model="gpt-4-turbo-preview", temperature=0, max_tokens=256)
embed_model = OpenAIEmbedding()
logging.basicConfig(stream=sys.stdout, level=logging.INFO)
logging.getLogger().addHandler(logging.StreamHandler(stream=sys.stdout))

st.markdown(
    """
            <style>
            #MainMenu {visibility: hidden;}
            footer {visibility: hidden;}
            header {visibility: hidden;}
            
    .reportview-container h1 {
        font-size: 1.5em;
        color: #0e3b62; 
        text-align: center;
        padding: 1em;
        background: #f1f1f1; 
        border-radius: 10px;
        border: 1px solid #ccc;
        box-shadow: 2px 2px 12px rgba(0, 0, 0, 0.1);
    }
</style>
""",
    unsafe_allow_html=True,
)


st.write(
    "<style>div.block-container{padding-top:0rem;}</style>", unsafe_allow_html=True
)

# Streamlit UI
st.title("Q&A System with Llama Index")

# Specify the directory containing the CSV files
input_directory = "Dataset"
# Create an instance of SimpleDirectoryReader pointing to the input directory
documents_reader = SimpleDirectoryReader(input_directory)
# Load the documents (CSV files in this case)
documents = documents_reader.load_data()
# print(documents)

node_parser = SimpleNodeParser.from_defaults(
    separator=" ",
    chunk_size=1024,
    chunk_overlap=20,
    paragraph_separator="\n\n\n",
    secondary_chunking_regex="[^,.;。]+[,.;。]?",
    tokenizer=tiktoken.encoding_for_model("gpt-4-turbo-preview").encode,
)
prompt_helper = PromptHelper(
    context_window=4096, num_output=256, chunk_overlap_ratio=0.1, chunk_size_limit=None
)

service_context = ServiceContext.from_defaults(
    llm=llm,
    embed_model=embed_model,
    node_parser=node_parser,
    prompt_helper=prompt_helper,
)

# Create an index from the documents
index = VectorStoreIndex.from_documents(documents, service_context=service_context)
# Define a custom prompt for trading signal transformation
template = (
    "Transform the trading signal input into a structured JSON format as shown in the examples below. \n"
    "Given this information, please answer the question and each answer should start with code word AI Demos: {query_str}\n"
)
qa_template = PromptTemplate(template)

# Create a query engine from the index
query_engine = index.as_query_engine(text_qa_template=qa_template)

# Adjust your query to fit the new format. Example:
# input_signal = "HIGH RISK SPX here 01/02 4730P Avg. 3.30"
# response = query_engine.query(input_signal)
# print(response)

# User input
user_query = st.text_input("Enter your question:", "")

# Query the engine
# response = query_engine.query("HIGH RISK SPX here 01/02 4730P Avg. 3.30")
# print(response)

if st.button("Get Answer"):
    # Query the engine
    response = query_engine.query(user_query)
    response_text = (
        response.response if hasattr(response, "response") else "No valid response."
    )
    st.write(response_text)

```
requirements.ttx:
```
llama-index
llama-index-core
llama-index-readers-file
llama-index-llms-openai
llama-index-embeddings-openai
openai==1.13.3
sentence_transformers==2.4.0
streamlit==1.31.1
isort
ruff 
black
python-dotenv
tiktoken
```

### Relevant Logs/Tracbacks

```shell
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 535, in _run_script
    exec(code, module.__dict__)
File "/Users/ankushsingal/Desktop/llamaindex/app.py", line 6, in <module>
    from llama_index.core import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/__init__.py", line 13, in <module>
    from llama_index.callbacks.global_handlers import set_global_handler
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/callbacks/__init__.py", line 7, in <module>
    from .token_counting import TokenCountingHandler
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/callbacks/token_counting.py", line 6, in <module>
    from llama_index.utilities.token_counting import TokenCounter
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/utilities/token_counting.py", line 6, in <module>
    from llama_index.llms import ChatMessage, MessageRole
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/llms/__init__.py", line 3, in <module>
    from llama_index.llms.anyscale import Anyscale
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/llms/anyscale.py", line 9, in <module>
    from llama_index.llms.openai import OpenAI
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/llms/openai/__init__.py", line 1, in <module>
    from llama_index.llms.openai.base import AsyncOpenAI, OpenAI, SyncOpenAI, Tokenizer
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/llms/openai/base.py", line 16, in <module>
    from llama_index.core.base.llms.types import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/__init__.py", line 19, in <module>
    from llama_index.core.indices import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/indices/__init__.py", line 4, in <module>
    from llama_index.core.indices.composability.graph import ComposableGraph
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/indices/composability/__init__.py", line 4, in <module>
    from llama_index.core.indices.composability.graph import ComposableGraph
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/indices/composability/graph.py", line 7, in <module>
    from llama_index.core.indices.base import BaseIndex
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/indices/base.py", line 12, in <module>
    from llama_index.core.ingestion import run_transformations
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/__init__.py", line 2, in <module>
    from llama_index.core.ingestion.pipeline import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/pipeline.py", line 31, in <module>
    from llama_index.core.ingestion.api_utils import get_client
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/api_utils.py", line 20, in <module>
    from llama_index.core.ingestion.data_sources import (
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/data_sources.py", line 340, in <module>
    ConfigurableDataSources = build_conifurable_data_source_enum()
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/llama_index/core/ingestion/data_sources.py", line 117, in build_conifurable_data_source_enum
    DataSource(
File "/Users/ankushsingal/miniconda3/envs/snakes/lib/python3.10/site-packages/pydantic/v1/main.py", line 341, in __init__
    raise validation_error
```
```

**Comment by logan-markewich** [comment]
@andysingal try with a fresh venv

In a new terminal
```
pip uninstall llama-index  # remove any global install
python -m venv venv
source venv/bin/activate
pip install llama-index --upgrade --no-cache-dir --force-reinstall
```

**Comment by namedgraph** [comment]
@logan-markewich can you explain _why_ a reinstall is necessary?

**Comment by logan-markewich** [comment]
@namedgraph the switch from v0.9.x to v0.10.x introduced namespaced packages. Any remnants of the old v0.9.x causes some weird issues, and the above is one of the issues I've seen it cause

Trying with a fresh venv should fix the issue in nearly every case I've encountered (especially considering this works fine for me locally and in a fresh Google colab)

**Comment by nataliachodelski** [comment]
TLDR: 
I was unable to resolve it with the steps shared in this post, but i was able to resolve this bug by switching to `llama_index==0.10.19`. Also after using 0.10.19, I can now use the most recent version of llama_index and the bug no longer occurs.

---- 
Details:

In my case, i'm running a python microservice in a Kubernetes node in Docker, and each time the container is updated, it reinstalls all python packages from scratch. When I tried adding  `pip uninstall llama-index` to my Dockerfile before the command to install packages from my `requirements.txt` file, the response was `→ WARNING: Skipping llama-index as it is not installed`, so in my case this issue is not from any remnants of an older version - its happening in a 'fresh' install on my container. 

**However, I realized this error is happening in only 1 of my 2 microservices that both use `llama_index`.**  When I tried changing the versions of 3 llama_index packages in the 'bugged' service to the versions being used in the working service, it resolved the issue for me!

I experienced this bug when i was using: 
```
llama_index==(0.10.20, 0.10.23, 0.10.24, 10 0.10.26) 
llama-index-vector-stores-lantern==0.1.2 
llama-index-readers-s3==0.1.5 
```

However, when i updated my versions to this combo (which is what my other service was using), it went away:
```
llama_index==0.10.19. (<- I think changing llama_index to this version was what fixed it for me)
llama-index-vector-stores-lantern==0.1.2
llama-index-readers-s3==0.1.3
```

Even more interesting, after I updated my requirements file to the above listed versions which seemed to 'fix' the issue, I then tried changed the versions of llama_index & llama-index-readers-s3 back to the latest versions of each (0.10.26 & 0.1.5), and the project now runs without the error. **The bug seems to have been fixed by installing llama_index 0.10.19 and I can now use later versions again.**     I will report here if I experience it again.





This is my environment & hardware in case its relevant:
- occurred running my project on both a 2021 M1 Macbook Pro & a 2022 M2 Macbook Air (ARM chips)
- using python 3.12.0
- error reproducible  upon upon running python project (in my project, the error was coming from a Class that imports llama_index which was in turn imported by main.py)
- experienced bug in a Kubernetes node running in Docker 
- I also switched to a different computer and the issue continued to occur upon running container in docker on a different laptop!

**Comment by nataliachodelski** [comment]
@logan-markewich, I suspect this issue might be occurring for people using ARM macbooks. A new teammate just got setup on my project and encountered this issue as well, and he is using a 2022 M2 macbook air.   I have also been repeatedly experiencing this issue on a 2021 M1 Macbook.   

@andysingal - what kind of computer & processor type were using when you experienced this issue?  

So I've experienced this issue about 5 separate times over about 10 days.  Each time I've been able to resolve it by changing the specific version of llama_index being installed from my `requirements.txt` file. Sometimes downgrading form. 0.10.26 to 0.10.19 fixes it, other times it'll occur when i'm using 0.10.19 and switching  0.10.24 fixes it.    When my new teammate just ran into this issue, he was initally using 0.10.20, and he tried 0.10.19, 0.10.26, and then 0.10.24.  The switch from .26 -> .24 seemed to resolve it for him this time.

**Comment by logan-markewich** [comment]
@nataliachodelski I have yet to encounter this issue even once on my m2 max 👀 

Tbh, I just use venv and everything works fine. I'm constantly installing new versions as well as part of my day to day work

**Comment by tsantra** [comment]
after updating llama-index to 0.10.28 i am not getting this error.

**Comment by logan-markewich** [comment]
Going to close out this issue as tbh there is nothing actionable left here. To feel free to open a new issue if something comes up.
